src/layout/vertical.lua

```diff
@@ -0,0 +1,47 @@
+local VerticalLayout = class('VerticalLayout', ikkuna.Layout)
+
+function VerticalLayout:initialize(options)
+	local options = options or {}
+	self.updatesEnabled = options.updatesEnabled or true
+	self.fitParent = options.fitParent or false
+	self.childSpacing = options.childSpacing or 5
+end
+
+function VerticalLayout:setParent(parent)
+	self.parent = parent
+end
+
+function VerticalLayout:setFitParent(fit)
+	self.fitParent = fit
+	self:update()
+end
+
+function VerticalLayout:updateInternal()
+	-- TODO: Take padding and margins into account
+	if self.fitParent then
+		local parent = self.parent
```

Comment:
parent, width and position variables are the same in both branches, thus could be extracted above the if-statement 🤔